### PR TITLE
[MIG] base_ubl: Migration to 14.0

### PR DESCRIPTION
**Pull request by phuctranfxvn**
Requires modules:
- `account_tax_unece` (https://github.com/OCA/community-data-files/pull/99)
- `uom_unece` (https://github.com/OCA/community-data-files/tree/14.0)

### CODE REVIEW

**Comment by StefanRijnhart** [comment]
I see you tried to add a filter on `xml` files in the prettier config, but the UBL xsd files were reformatted regardless (in the preceding "black, isort, prettier" commit). Is that something that should be prevented?

**Comment by phuctranfxvn** [comment]
> I see you tried to add a filter on `xml` files in the prettier config, but the UBL xsd files were reformatted regardless (in the preceding "black, isort, prettier" commit). Is that something that should be prevented?

Thanks @StefanRijnhart
I was added to prevent the checking of xsd files on running travis. But yes, agree with you that the file should not be modifed in "black, isort, prettier". I have force-pushed the changes

**Comment by bealdav** [comment]
Hi @phuctranerp, thanks for this migration and additional unit test on invoice.
I ask myself if we should make some changes asked in https://github.com/OCA/edi/issues/246#issuecomment-733141789 in this PR or made them after merged this PR ?
any idea @simahawk ?

**Comment by simahawk** [comment]
> I ask myself if we should make some changes asked in [#246 (comment)](https://github.com/OCA/edi/issues/246#issuecomment-733141789) in this PR or made them after merged this PR ?
> any idea @simahawk ?

I comment on #246

**Comment by StefanRijnhart** [comment]
Can I conclude that the adoption of this module in the 14.0 series will not be blocked by the strategy as laid out in #246? I base this conclusion mainly on Alexis' comment https://github.com/OCA/edi/issues/246#issuecomment-791350434

File: base_ubl/__manifest__.py [hunk 1]
```diff
@@ -3,7 +3,7 @@
 
 {
     "name": "Base UBL",
-    "version": "13.0.2.0.1",
+    "version": "14.0.2.0.1",
```
Comment (dsolanki-initos) [inline base_ubl/__manifest__.py]: Refer guideline for module version:
https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-14.0

Comment (phuctranfxvn) [reply base_ubl/__manifest__.py]: fixed, thanks

Comment (StefanRijnhart) [reply base_ubl/__manifest__.py]: Please squash the related commit into the migration commit.

**Comment by bealdav** [comment]
@StefanRijnhart probably.
About using component, it's a major change that could be done or not (it requires efforts) but probably not in this PR.
I proposed to use mixin to split methods in this module according to which models they implies.
It could be a minor change and could be done here. But I don't want force it now. I just submit the idea.

File: base_ubl/__manifest__.py (hunk 1 above)
Comment (StefanRijnhart) [reply base_ubl/__manifest__.py]: @phuctranerp you resolved this comment but the separate commit for the version update is still present on the branch.

Comment (phuctranfxvn) [reply base_ubl/__manifest__.py]: @StefanRijnhart I am thinking if we should track the history of changes. Btw, I have squashed the commits. Thanks for your feedbacks

File: base_ubl/tests/test_ubl_generate.py [hunk 2]
```diff
@@ -29,3 +29,83 @@ def test_ubl_generate(self):
         self.env["base.ubl"]._ubl_add_customer_party(
             invoice.partner_id, False, "AccountingCustomerParty", xml_root, ns
         )
+
+    def create_test_invoice(
```
Comment (StefanRijnhart) [inline base_ubl/tests/test_ubl_generate.py:33]: Please credit Alexis for this code as it was originally added to account_tax_unece in https://github.com/OCA/community-data-files/pull/24. You can add

```
# © 2017-2020 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
```
at the top of this file.

Comment (StefanRijnhart) [reply base_ubl/tests/test_ubl_generate.py:33]: We'd like to get this merged. We'll create a superseding PR with this change.

Comment (rconjour) [reply base_ubl/tests/test_ubl_generate.py:33]: @phuctranerp created superseeding PR: https://github.com/OCA/edi/pull/361

**Comment by StefanRijnhart** [comment]
Continuing on the other PR. Thank you for your work @phuctranerp! Let's hope to get this merged soon.

**Pull request closed without merging**